flytekit/core/type_engine.py

```diff
@@ -234,6 +234,47 @@ def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], exp
             scalar=Scalar(generic=_json_format.Parse(cast(DataClassJsonMixin, python_val).to_json(), _struct.Struct()))
         )
 
+    def _fix_val_int(self, t: typing.Type, val: typing.Any) -> typing.Any:
+        if t == int:
+            return int(val)
+
+        if isinstance(val, list):
+            try:
+                if ListTransformer.get_sub_type(t) == int:
+                    return [int(i) for i in val]
```

Comment:
How about we replace `[int(i) for i in val]` with `list(map(int, val))`?

Reply:
Updated it, Thanks.